primefaces/src/main/java/org/primefaces/component/rating/RatingRenderer.java

```diff
@@ -142,7 +141,7 @@ protected void encodeInput(FacesContext context, Rating rating, String id, Strin
         writer.writeAttribute("min", "0", null);
         writer.writeAttribute("max", rating.getStars(), null);
         writer.writeAttribute("autocomplete", "off", null);
-        writer.writeAttribute("value", Objects.toString(value, "0"), null);
+        writer.writeAttribute("value", LangUtils.isBlank(value) ? "0" : value, null);
```

Comment:
We could introduce a new method here: `LangUtils#toString(value, blankDefault)`

Reply:
I added `defaultIfBlank` to match commong Lang StringUtils.